ext/pgsql/pgsql.c

```diff
@@ -3211,18 +3289,19 @@ PHP_FUNCTION(pg_copy_from)
 					if (UNEXPECTED(!tmp)) {
 						return;
 					}
-					query = (char *)emalloc(ZSTR_LEN(tmp) + 2);
-					strlcpy(query, ZSTR_VAL(tmp), ZSTR_LEN(tmp) + 2);
-					if (ZSTR_LEN(tmp) > 0 && *(query + ZSTR_LEN(tmp) - 1) != '\n') {
-						strlcat(query, "\n", ZSTR_LEN(tmp) + 2);
+					zend_string *zquery = zend_string_alloc(ZSTR_LEN(tmp) + 2, false);
+					memcpy(ZSTR_VAL(zquery), ZSTR_VAL(tmp), ZSTR_LEN(tmp) + 2);
```

Comment:
This is wrong, no?
You're copying 1 byte outside of the tmp string. It should be + 1 instead of + 2.

Reply:
true